astropy/convolution/utils.py

```diff
@@ -201,28 +201,17 @@ def discretize_model(model, x_range, y_range=None, mode="center", factor=10):
         raise ValueError("y_range must be specified for a 2D model")
     if ndim == 1 and y_range is not None:
         raise ValueError("y_range should not be input for a 1D model")
-    if mode == "center":
-        if ndim == 1:
-            return discretize_center_1D(model, x_range)
-        if ndim == 2:
-            return discretize_center_2D(model, x_range, y_range)
-    elif mode == "linear_interp":
-        if ndim == 1:
-            return discretize_linear_1D(model, x_range)
-        if ndim == 2:
-            return discretize_bilinear_2D(model, x_range, y_range)
-    elif mode == "oversample":
-        if ndim == 1:
-            return discretize_oversample_1D(model, x_range, factor)
-        if ndim == 2:
-            return discretize_oversample_2D(model, x_range, y_range, factor)
-    elif mode == "integrate":
-        if ndim == 1:
-            return discretize_integrate_1D(model, x_range)
-        if ndim == 2:
-            return discretize_integrate_2D(model, x_range, y_range)
-    else:
-        raise ValueError("Invalid mode for discretize_model.")
+    args = [model, x_range]
+    if ndim == 2:
+        args.append(y_range)
+    if mode == "oversample":
+        args.append(factor)
+    if mode == "linear_interp":
+        return (discretize_linear_1D if ndim == 1 else discretize_bilinear_2D)(*args)
+    try:
+        return globals()[f"discretize_{mode}_{ndim}D"](*args)
```

Comment:
I like the simplification overall, but less so the call to `globals`. What if you made a dictionary with tuple keys?

```suggestion
        return _discretization_ops[(mode, ndim)](*args)
```

Reply:
Having a separate dictionary would result in more verbose code, but there would be no need to handle `mode == "linear_interp"` differently from the rest, so the tradeoff might be worth it. But I'd like feedback from more reviewers before making changes.

Reply:
I like @nstarman's suggestion.  I greatly prefer the clarity of explicit code rather than having to refer to a scope that's not easily understood at a glance.